### PR TITLE
VZ Events: Real-time main-lane dissociation

### PR DESCRIPTION
**Pull request by sergiogcx**
Closes https://github.com/cityofaustin/atd-data-tech/issues/2210

Integration Tests:
https://github.com/cityofaustin/atd-data-tech/issues/3646

### CODE REVIEW

File: .circleci/aws-lambda-helper.sh [hunk 1]
```diff
@@ -162,7 +162,7 @@ function clean_up {
 #
 function deploy_event_functions {
   MAIN_DIR=$PWD
-  for FUNCTION in $(find atd-events -type d -mindepth 1 -maxdepth 1);
+  for FUNCTION in $(find atd-events -type d -mindepth 1 -maxdepth 1 ! -iname "tests");
```
Comment (sergiogcx) [inline .circleci/aws-lambda-helper.sh:165]: This like enforces exclusion of the test folder

File: atd-vzd/triggers/find_cr3_mainlane_crash.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    SELECT atc.*
+    FROM atd_txdot_crashes AS atc
+            INNER JOIN cr3_mainlanes AS cr3m ON (
+            atc.position && cr3m.geometry
+            AND ST_Contains(
+                    ST_Transform(
+                            ST_Buffer(
+                                    ST_Transform(cr3m.geometry, 2277),
+                                    1,
+                                    'endcap=flat join=round'
```
Comment (mateoclarke) [inline atd-vzd/triggers/find_cr3_mainlane_crash.sql:15]: 👀

File: atd-vzd/views/view_crashes_mainlane_noncr3.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+CREATE OR REPLACE VIEW view_crashes_mainlane_noncr3 AS
```
Comment (mateoclarke) [inline atd-vzd/views/view_crashes_mainlane_noncr3.sql:1]: are we using these two mainline views in the UI? or are they for Franks and reporting? just curious.

**Comment by mateoclarke** [comment]
@sergiogcx, I'd love if you had some time to pair with me and talk me through running the test suite locally. Would love to grow my knowledge from you that way.

**Comment by mateoclarke** [comment]
@sergiogcx, I'm going to go ahead and merge this to master to include in the `1.13.0`. My apologies for not getting to this earlier this afternoon.

File: atd-events/crash_update_jurisdiction/app.py [hunk 4]
```diff
@@ -1,125 +1,230 @@
...
-    # Prep Hasura query
-    HEADERS = {
-        "Content-Type": "application/json",
-        "X-Hasura-Admin-Secret": HASURA_ADMIN_SECRET,
-    }
+def is_crash_in_jurisdiction(crash_id: int) -> str:
+    """
+    Attempts to find the jurisdiction of a crash by it's ID
+    :param int crash_id: The crash_id to be evaluated
+    :return str: The location_id string
```
[the remark this answers is not in the document]
Comment (sergiogcx) [reply atd-events/crash_update_jurisdiction/app.py:90]: Ah, good call. Thanks for catching it. I must have refactored the function or something and I did not care to 2x check the comments

File: atd-vzd/views/view_crashes_mainlane_noncr3.sql (hunk 3 above)
Comment (sergiogcx) [reply atd-vzd/views/view_crashes_mainlane_noncr3.sql:1]: They are not currently used or tracked in Hasura, but I created them to document how Frank finds main-lane crashes.

Comment (mateoclarke) [reply atd-vzd/views/view_crashes_mainlane_noncr3.sql:1]: nice, thanks!

**Pull request merged**